### PR TITLE
WIP: Twu view

### PR DESCRIPTION
**Pull request by bdolor**
This PR closes issue: [issue DM-1126]

Includes tests? [N]
Updated docs? [Y]

Proposed changes:
- TWU View Interface
- includes detailed view with tabs
- edit view with summary, opportunity, addenda and history tabs
- back-end changes for subscription/watch functionality
- cards for opportunity page 

Additional notes:
- dashboard view and finished FE watch functionality will be in a separate PR `refactor-helper-interfaces`

### CODE REVIEW

File: src/back-end/lib/db/opportunity/team-with-us.ts [hunk 1]
```diff
@@ -1,5 +1,10 @@
 import { generateUuid } from "back-end/lib";
-import { Connection, Transaction, tryDb } from "back-end/lib/db";
+import {
+  Connection,
+  RawSWUOpportunitySubscriber,
```
Comment (IanFonzie) [inline src/back-end/lib/db/opportunity/team-with-us.ts:4]: SWU Reference

File: src/back-end/lib/db/opportunity/team-with-us.ts [hunk 2]
```diff
@@ -442,34 +499,34 @@ function processForRole<T extends RawTWUOpportunity | RawTWUOpportunitySlim>(
...
+    ? valid(await rawTWUOpportunitySlimToTWUOpportunitySlim(connection, result))
+    : valid(null);
+});
+
+async function isSubscribed(
+  connection: Connection,
+  oppId: Id,
+  userId: Id
+): Promise<boolean> {
+  return !!(await connection<RawSWUOpportunitySubscriber>(
```
Comment (IanFonzie) [inline src/back-end/lib/db/opportunity/team-with-us.ts:524]: SWU Generic Type Param

File: src/front-end/typescript/lib/pages/opportunity/code-with-us/lib/views/edit-tab-header.tsx [hunk 3]
```diff
@@ -73,7 +73,6 @@ const EditTabHeader: component.base.View<Props> = ({
           <h3 className="mb-2">
             Code With Us:&nbsp;
             <Link
-              newTab
```
Comment (IanFonzie) [inline src/front-end/typescript/lib/pages/opportunity/code-with-us/lib/views/edit-tab-header.tsx:76]: Was this intentional?

File: src/front-end/typescript/lib/pages/opportunity/list.tsx [hunk 4]
```diff
@@ -176,15 +196,16 @@ const init: component_.page.Init<
...
   const [typeFilterState, typeFilterCmds] = Select.init({
     errors: [],
     child: {
       value: null,
       id: "opportunity-filter-type",
       options: adt("options", [
         { label: "Code With Us", value: "cwu" },
-        { label: "Sprint With Us", value: "swu" }
+        { label: "Sprint With Us", value: "swu" },
+        { label: "Sprint With Us", value: "twu" }
```
Comment (IanFonzie) [inline src/front-end/typescript/lib/pages/opportunity/list.tsx:208]: Label text, which I think Steve pointed out today

**Review by IanFonzie — CHANGES_REQUESTED** [review]

File: src/front-end/typescript/lib/pages/opportunity/code-with-us/lib/views/edit-tab-header.tsx (hunk 3 above)
Comment (bdolor) [reply src/front-end/typescript/lib/pages/opportunity/code-with-us/lib/views/edit-tab-header.tsx:76]: 100% intentional

File: src/front-end/typescript/lib/pages/opportunity/list.tsx (hunk 4 above)
Comment (bdolor) [reply src/front-end/typescript/lib/pages/opportunity/list.tsx]: complete

File: src/back-end/lib/db/opportunity/team-with-us.ts (hunk 2 above)
Comment (bdolor) [reply src/back-end/lib/db/opportunity/team-with-us.ts]: complete

File: src/back-end/lib/db/opportunity/team-with-us.ts (hunk 1 above)
Comment (bdolor) [reply src/back-end/lib/db/opportunity/team-with-us.ts]: complete

**Review by IanFonzie — APPROVED** [review]

**Pull request merged**